### ISSUE TITLE
Very weird visual glitches, not a single thing to find in log files regarding this bug. 1.19.2 Forge. 

### ISSUE THREAD

**Issue by DigitalWendigo** [FiniteReality/embeddium#193]
### Bug Description

This is an odd one that I can't figure out, since initially I thought it had something to do with transparent face sorting, but now it occurs with blocks that aren't technically transparent (I think..?) 
It occurs with both vanilla blocks and modded blocks, all seemingly at random, and with no reliable way to actually pinpoint when it will happen. I've provided a few steps below to try and replicate it, but it seems to happen truly at random. Sometimes even only after a few days of playing in the world, with and without other mods. 
Pictures will describe it best. 
 
![2024-01-21_23 22 24](https://github.com/embeddedt/embeddium/assets/103202053/87f6b830-6934-4410-9392-4e02dd3d0d91)
![2024-01-26_15 22 27](https://github.com/embeddedt/embeddium/assets/103202053/fb4dd296-4c3a-4207-ab74-9f925dd428de)
![2024-01-27_13 27 23](https://github.com/embeddedt/embeddium/assets/103202053/418c4a82-6918-49e7-bbc0-ccb90b9c5870)

Let me know if there is something else I need to provide. 
Oddly enough, this did not happen when I used Rubidium, even if that came with its own set of bugs.
Cheers!

### Reproduction Steps

1. Install Embeddium.  
2. Install any other mods alongside it. (Optional, visual glitches occur even with just Embeddium despite the latest provided log showing a lost of mods used.) 
3. Enter the world and fly around, entering both new chunks, and going back to old ones. 
4. Eventually encounter issues as shown in the provided pictures. 

### Log File

[latest.log](https://github.com/embeddedt/embeddium/files/14141769/latest.log)

**Comment by embeddedt** [comment]
If it happens without any other mods I would suggest ensuring your GPU drivers are up-to-date. You could also try disabling block face culling in video settings and see if that makes a difference.

Aside from that, I'm not really sure how else to help... this is not an issue that anyone else has reported, so I suspect it's a problem with your particular system.

**Comment by DigitalWendigo** [comment]
Alright, I'll see if disabling block face culling changes anything. 
Unfortunately my drivers are all up to date, so I really hope it's not a gpu issue since the whole setup is relatively new. 
Thanks for the quick reply!

**Comment by DigitalWendigo** [comment]
Hey there! 
So I've got an update on the issue. 
I've since removed a lot of other mods from my pack to try and figure out if perhaps a mod messes something up in existing worlds, and now am trying to see if it still occurs in new worlds. 
Unfortunately, this still happens. 
What's more is that a new issue has occured, which happens if one travels fast while flying in creative, at which point chunks don't generate fast enough, and rendering breaks entirely. 
Neither my gpu nor cpu seem to be stressing much while exploring new chunks, so I'm curious as to what the cause is. 
I've even tried disabling block face culling, but that doesn't seem to do anything about the issue. 
Here's a screenshot of the latest issue. 
![2024-02-03_15 06 29](https://github.com/embeddedt/embeddium/assets/103202053/67555e85-d975-45c3-8217-3971e88f4215)

Here are also my system specs, in case the gpu is completely incompatible with embeddium. 
GPU: NVIDIA Quadro 2000 
CPU: Intel Xeon CPU X5570 @ 2.93GHz 
RAM: 16GB DDR (4G assigned to minecraft.) 
Java Arguments: -XX:+UseG1GC -Xmx4G -Xms4G -Dsun.rmi.dgc.server.gcInterval=2147483646 -XX:+UnlockExperimentalVMOptions -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
Harddrive: Seagate ST500DM002-1SB10A 

If you need any more information or testing, let me know! 

Edit: Placing down any kind of block in the visually glitched out 4x4 chunk area does not fix it, and any blocks placed in it, including entities that walk into it become completely invisible.

**Comment by embeddedt** [comment]
I am not really sure how Embeddium would cause this. The symptom in that screenshot looks more like a server-side performance mod causing chunks not to load correctly; the renderer cannot show chunks that aren't loaded. (The game runs a server even in singleplayer.)

It would be useful to do all testing with only Embeddium installed, and no other mods, and try to determine a reliable way to reproduce the issue. Fixing it will be difficult if I can't replicate it on my end.

**Comment by DigitalWendigo** [comment]
Alrighty! 
I'll remove them one by one to find the culprit. 
I may have a loose idea of which this may be cause by, but only testing will confirm or deny it.

**Comment by DigitalWendigo** [comment]
Found the culprit! 
It's a mod called Better Chunk Loading. 
Removing this mod and making a new world seems to resolve both the incorrect rendering of certain blocks, and visually missing chunks. 
Interestingly enough, if you remove this mod and keep playing on an existing world, even with just embeddium and its addons, the issue persists, likely due to some corruption with the world. 
Playing in a world with only Better Chunk Loading does not replicate the issue, so it's some kind of incompatibility between the two mods. 
I'd still favor Embeddium over the other mod, so I'll be going forward without Better Chunk Loading.

Might be worth the time to mention that Embeddium isn't compatible with Better Chunk Loading to avoid future bug reports regarding the mentioned mod. (Better Chunk Loading is also closed source, so I'd imagine it would be difficult to find a fix for it.) 

Cheers!

**Comment by embeddedt** [comment]
I will make a note of this on the compatibility page. Thanks for your help.

**Comment by DigitalWendigo** [comment]
And thank you for your efforts to make minecraft run smoother! 
Have a great day friend!

**Comment by someaddons** [comment]
for the record better chunk loading is visible source no closed source.
It seems that embeddium has problems with its chunk tracking system, seeing as it seems to run into issues with both chunk loading(which changes the shape of chunks loaded on serverside, so they may load in a different order) and chunk sending( which delays sending chunks and sends them few at a time, sorting by closest)
@embeddedt since it works fine on vanilla for both my mods it I'd argue that this is a bug you should be fixing on your side instead of just silently marking it as incompatible
